README-config.md

```diff
@@ -0,0 +1,151 @@
+  hosts: ipaserver
+  become: true
+  tasks:
+    - name: set ipadefaultloginshell and ipamaxusernamelength
+      ipaconfig:
+        ipaadmin_password: password
+        ipadefaultloginshell: /bin/bash
+        ipamaxusernamelength: 64
+```
+
+
+Variables
+=========
+
+ipauser
+-------
+
+**General Variables:**
+
+Variable | Description | Required
+-------- | ----------- | --------
+`ipaadmin_principal` | The admin principal is a string and defaults to `admin` | no
+`ipaadmin_password` | The admin password is a string and is required if there is no admin ticket available on the node | no
+`ipaadmin_principal` |  The admin principal (default: admin) | false
+`ipaadmin_password` |  The admin password | false
+`ipamaxusernamelength` \| `maxusername` |  Set the maximum username length (1 to 255) | false
+`ipahomesrootdir` \| `homedirectory` |  Set the default location of home directories | false
+`ipadefaultloginshell` \| `defaultshell` |  Set the default shell for new users | false
+`ipadefaultprimarygroup` \| `defaultgroup` |  Set the default group for new users | false
+`ipadefaultemaildomain` \| `emaildomain` |  Set the default e-mail domain | false
+`ipasearchtimelimit` \| `searchtimelimit` |  Set maximum amount of time (seconds) for a search -1 to 2147483647 (-1 or 0 is unlimited) | false
+`ipasearchrecordslimit` \| `searchrecordslimit` |  Set maximum number of records to search -1 to 2147483647 (-1 or 0 is unlimited) | false
+`ipausersearchfields` \| `usersearch` |  Set list of fields to search when searching for users | false
+`ipagroupsearchfields` \| `groupsearch` |  Set list of fields to search in when searching for groups | false
+`ipamigrationenabled` \| `enable-migration` |  Enable migration mode (choices: True, False ) | false
+`ipagroupobjectclasses` \| `groupobjectclasses` |  Set default group objectclasses (list) | false
+`ipauserobjectclasses` \| `userobjectclasses` |  Set default user objectclasses (list) | false
+`ipapwdexpadvnotify` \| `pwdexpnotify` |  Set number of days's notice of impending password expiration (0 to 2147483647) | false
+`ipaconfigstring` \| `ipaconfigstring` |  Set extra hashes to generate in password plug-in (choices:`AllowNThash`, `KDC:Disable Last Success`, `KDC:Disable Lockout`, `KDC:Disable Default Preauth for SPNs`) | false
+`ipaselinuxusermaporder` \| `ipaselinuxusermaporder` | Set ordered list in increasing priority of SELinux users | false
+`ipaselinuxusermapdefault` \| `ipaselinuxusermapdefault`  |  Set default SELinux user when no match is found in SELinux map rule | false
+`ipakrbauthzdata` \| `pac-type` |  set default types of PAC supported for services (choices: `MS-PAC`, `PAD`, `nfs:NONE`)
+`ipauserauthtype` \| `user-auth_type` |  set default types of supported user authentication (choices: `password`, `radius`, `otp`, `disabled`) | false
+`ipadomainresolutionorder` \| `domain-resolution_order` | Set list of domains used for short name qualification | false
```

Comment:
There is an 'underline' between 'resolution' and 'order', when there should be a 'dash': `domain-resolution-order`.
Please, use the CLI name, the second one, as the default variable name.

Reply:
hmm, the cli claims the option is domain-resolution-order, but the online api documentation ($IPASERVER/ipa/ui/#/p/apibrowser/type=command&name=config_mod)  says "CLI option name  --domain-resolution_order"    so theres a bug somewhere.... :)

Reply:
Please do not use dashed in variables at all. Use underscores instead.